src/ElasticsearchEngine.php

```diff
@@ -24,16 +24,19 @@ class ElasticsearchEngine extends Engine
      */
     protected $elastic;
 
+    protected $type = 'doc';
```

Comment:
For future compatibility, consider defaulting `$type` to `_doc` instead of `doc`. See notes under Elasticsearch 7.x in the [roadmap](https://www.elastic.co/guide/en/elasticsearch/reference/6.0/removal-of-types.html#_schedule_for_removal_of_mapping_types).

Reply:
Document mapping type name can't start with '_'.

Reply:
Oh, interesting. My bad, they must reserve it for internal use. Thanks for giving it a shot.

Reply:
Looks like that might be a bug that's fixed in 6.2: https://github.com/elastic/elasticsearch/pull/27816

Found it in this discussion:
https://discuss.elastic.co/t/cant-use-doc-as-type-despite-it-being-declared-the-preferred-method/113837